include/zephyr/drivers/stepper_motor_controller.h

```diff
@@ -0,0 +1,257 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_STEPPER_MOTOR_CONTROLLER_H
+#define ZEPHYR_INCLUDE_STEPPER_MOTOR_CONTROLLER_H
+
+/** \addtogroup stepper_motor_controller
+ *  @{
+ */
+
+/**
+ * @file
+ * @brief Public API for Stepper Motor Controller
+ *
+ */
+
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Stepper Motor Position Format required while setting or getting stepper motor positions.
+ */
+enum micro_step_resolution {
+	FULL_STEP,
+	MICRO_STEP_2,
+	MICRO_STEP_4,
+	MICRO_STEP_8,
+	MICRO_STEP_16,
+	MICRO_STEP_32,
+	MICRO_STEP_64,
+	MICRO_STEP_128,
+	MICRO_STEP_256,
+};
+
+/**
+ * @typedef stepper_motor_controller_set_steps_to_move_t
+ * @brief Stepper Motor Controller shall move the motor with given steps
+ */
+typedef int32_t (*stepper_motor_controller_set_steps_to_move_t)(const struct device *dev,
+								int32_t steps);
```

Comment:
```suggestion
typedef int32_t (*stepper_motor_controller_move_t)(const struct device *dev,
								int32_t steps);
```
I would suggest to omit the steps in the typedef name, to keep it short. That it is about to move in steps is already given by the name of the 2nd parameter.

This would also affect the api function below.

Reply:
But the controller cannot move  ;-)

Reply:
`stepper_motor_move`? Almost tempted to drop "motor" as well, keep it more compact, `drivers/stepper`, `stepper_move` `CONFIG_STEPPER_INIT_PRIORITY` and so on... don't know would there be any ambiguity? "stepper motor controller" over anything feels a bit heavy